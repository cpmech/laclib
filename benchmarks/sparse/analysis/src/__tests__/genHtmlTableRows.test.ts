import { genHtmlTableRows } from '../genHtmlTableRows';
import { IHtmlColData } from '../types';

describe('genHtmlTableRows', () => {
  it('generates row data', () => {
    const columns: IHtmlColData[] = [
      {
        label: '',
        ana: 'ana.0',
        fac: 'fac.0',
        sol: 'sol.0',
        tot: 'tot.0',
        normAx: 'nax.0',
        relErr: 'rer.0',
      },
      {
        label: '',
        ana: 'ana.1',
        fac: 'fac.1',
        sol: 'sol.1',
        tot: 'tot.1',
        normAx: 'nax.1',
        relErr: 'rer.1',
      },
    ];
    const rows = genHtmlTableRows(columns, true);
    expect(rows).toMatchSnapshot();
  });
});
