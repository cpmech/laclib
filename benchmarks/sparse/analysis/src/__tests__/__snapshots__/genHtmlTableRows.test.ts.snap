// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genHtmlTableRows generates row data 1`] = `
"
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Analyze</th>
    ana.0
  </tr>
  <tr>
    ana.1
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Factorize</th>
    fac.0
  </tr>
  <tr>
    fac.1
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Solve</th>
    sol.0
  </tr>
  <tr>
    sol.1
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Total</th>
    tot.0
  </tr>
  <tr>
    tot.1
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Linf(A.x)</th>
    nax.0
  </tr>
  <tr>
    nax.1
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"2\\">Rel Error</th>
    rer.0
  </tr>
  <tr>
    rer.1
  </tr>"
`;
